staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service.go

```diff
@@ -158,10 +159,28 @@ func (g *gRPCService) Encrypt(plain []byte) ([]byte, error) {
 		return nil, err
 	}
 
-	request := &kmsapi.EncryptRequest{Plain: plain, Version: kmsapiVersion}
+	request := &kmspb.EncryptRequest{Plain: plain, Version: kmsapiVersion}
 	response, err := g.kmsClient.Encrypt(ctx, request)
 	if err != nil {
 		return nil, err
 	}
 	return response.Cipher, nil
 }
+
+// Check the status of KMS provider.
+func (g *gRPCService) Check() error {
+	ctx, cancel := context.WithTimeout(context.Background(), g.callTimeout)
```

Comment:
We probably need a different time-out for this.
This will be executed as part of the overall kube-apiservers' healthz check. Therefore, we can't block here for a long time - need to fail fast here. I would suggest a separate connection objects for the health client.

Reply:
will use `kmsPluginConnectionTimeout`

Reply:
I think that kmsPluginConnectionTimeout may be too long for a health check since health checks have to complete relatively quickly. The default is 3 seconds.